### PR TITLE
Noahcaldwell/batting stats view

### PR DESCRIPTION
**Pull request by Debob123**
# Overview
added wrc_plus to batting stats view, added playing time calcuations for batting stats and pitching stats

## Changes
added section to SQL query for pitching stats view and batting stats view to include playing time. Added WarData ORM and functions to read in warData table to batting stats view. Also added csv file containing war and wrcplus info to dbSetup/static/csv

## Proof of Life
(I don't remember what proof of life means)

### CODE REVIEW

**Comment by brendonnewt** [comment]
> # Overview
> added wrc_plus to batting stats view, added playing time calcuations for batting stats and pitching stats
> 
> ## Changes
> added section to SQL query for pitching stats view and batting stats view to include playing time. Added WarData ORM and functions to read in warData table to batting stats view. Also added csv file containing war and wrcplus info to dbSetup/static/csv
> 
> ## Proof of Life
> (I don't remember what proof of life means)

Proof of life means some evidence that the feature/bug-fix does what it is intended to do. So some sort of proof that it's been tested and works.

File: dbSetup/models/tables/__init__.py [hunk 1]
```diff
@@ -786,6 +786,11 @@ class Divisions(Base):
         CheckConstraint("division_active IN ('Y', 'N')", name="chk_division_active"),
     )
 
+
+
+
```
Comment (brendonnewt) [inline dbSetup/models/tables/__init__.py:791]: Why so many lines? 😭

File: dbSetup/services/create_battingstats_view.py [hunk 2]
```diff
@@ -62,6 +89,8 @@ def create_battingstats_view():
         fielding f ON b.playerID = f.playerID AND b.yearID = f.yearID AND a.teamID = b.teamID
     JOIN
         wobaweights w ON w.yearID = b.yearID
+    JOIN 
+        WarData br ON CONCAT(p.nameFirst, ' ', p.nameLast) = br.Name AND b.yearID = br.yearID
```
Comment (brendonnewt) [inline dbSetup/services/create_battingstats_view.py:93]: This isn't going to do what you want. Players can have the same name, please use playerID for all joins as those are unique.

File: dbSetup/services/create_battingstats_view.py [hunk 3]
```diff
@@ -1,7 +1,30 @@
...
+from sqlalchemy.orm import DeclarativeBase
+
+class Base(DeclarativeBase):
+    pass
+
+
+class WarData(Base):
+    __tablename__ = "WarData"
+    wardata_ID = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
+    Name = Column(String(255), nullable=False)
```
Comment (brendonnewt) [inline dbSetup/services/create_battingstats_view.py:23]: You don't have to include the name if all you're doing is joining it to the existing battingstatsview but you do need the playerID

File: dbSetup/services/batting_csv_service.py [hunk 4]
```diff
@@ -147,16 +147,21 @@ def split_csv(file_path, chunksize=CHUNK_SIZE):
...
 
-    chunks = list(split_csv(csv_file_path))
 
     # Use multiprocessing to process chunks
     with Pool(processes=NUM_PROCESSES) as pool:
-        results = pool.map(process_chunk, chunks)
+        batting_results = pool.map(process_chunk, batting_chunks)
+        
 
     # Aggregate results
```
Comment (brendonnewt) [inline dbSetup/services/batting_csv_service.py:166]: results variable needs to be changed to batting_results

File: dbSetup/services/create_battingstats_view.py [hunk 5]
```diff
@@ -32,6 +55,10 @@ def create_battingstats_view():
         ((b.b_H - (b.b_2B + b.b_3B + b.b_HR)) + (2 * b.b_2B) + (3 * b.b_3B) + (4 * b.b_HR)) / b.b_AB AS b_SLG,
         (((b.b_H - (b.b_2B + b.b_3B + b.b_HR)) + (2 * b.b_2B) + (3 * b.b_3B) + (4 * b.b_HR)) / b.b_AB) - (b.b_H / b.b_AB) AS b_ISO,
         (b.b_H - (b.b_2B + b.b_3B + b.b_HR)) AS b_b_1B,
+        (b.b_PA / (SELECT SUM(b2.b_PA) 
+               FROM batting b2 
+               WHERE b2.yearID = b.yearID AND b2.teamID = b.teamID)) * 100 AS b_Playing_Time
```
Comment (brendonnewt) [inline dbSetup/services/create_battingstats_view.py:60]: Missing a comma here

**Review by brendonnewt — CHANGES_REQUESTED** [review]
I can't check this because it has syntax errors. Please make sure to run the script and make sure it works so that I can check if the values are coming out as expected. I also left some other comments about things I would like to see changed.

**Comment by ickoxii** [comment]
> > # Overview
> > added wrc_plus to batting stats view, added playing time calcuations for batting stats and pitching stats
> > ## Changes
> > added section to SQL query for pitching stats view and batting stats view to include playing time. Added WarData ORM and functions to read in warData table to batting stats view. Also added csv file containing war and wrcplus info to dbSetup/static/csv
> > ## Proof of Life
> > (I don't remember what proof of life means)
> 
> Proof of life means some evidence that the feature/bug-fix does what it is intended to do. So some sort of proof that it's been tested and works.

i.e. screenshots, what you did to validate your implementation, etc.

File: dbSetup/services/create_battingstats_view.py [hunk 6]
```diff
@@ -1,7 +1,31 @@
...
+    SmallInteger,
+    String,
+)
+from sqlalchemy.orm import DeclarativeBase
+
+class Base(DeclarativeBase):
+    pass
+
+
+class WarData(Base):
```
Comment (cateburglar) [inline dbSetup/services/create_battingstats_view.py:20]: why are you defining wardata twice (in views/init and also here)? only define it once, and then import it as needed. additionally, your definition in views/init does not match this definition.

File: dbSetup/services/create_views_service.py
Comment (cateburglar) [inline dbSetup/services/create_views_service.py:1]: remove this file from this commit -- we deleted this file bc its unnecessary / repetitive

File: dbSetup/services/create_pitchingstats_view.py
Comment (cateburglar) [inline dbSetup/services/create_pitchingstats_view.py:1]: remove this file from your commit as well -- icko created a p_playing_time already

**Review by cateburglar — CHANGES_REQUESTED** [review]
see my comments

File: dbSetup/services/create_battingstats_view.py [hunk 7]
```diff
@@ -63,12 +69,12 @@ def create_battingstats_view():
...
+        WarData br ON br.playerID = b.playerID AND b.yearID = br.yearID
     WHERE 
-        -- exclude pitchers as best we can 
-        (  (b.b_AB + b.b_BB + b.b_HBP + b.b_SH + b.b_SF) > 0
-            OR (b.b_AB + b.b_BB + b.b_HBP + b.b_SH + b.b_SF) IS NULL)
-        AND
-        (f.position != 'P' OR f.position IS NULL);
+        (b.b_AB + b.b_BB + b.b_HBP + b.b_SH + b.b_SF) > 0  -- exclude pitchers
+        AND (f.position != 'P' OR f.position IS NULL);
+
```
Comment (cateburglar) [inline dbSetup/services/create_battingstats_view.py:77]: why did u change this? this is incorrect-- it needs to be the version that u deleted bc sometimes b_PA is null and that doesnt necessarily mean that that given player should be excluded

Comment (Debob123) [reply dbSetup/services/create_battingstats_view.py:79]: I'm sorry i didn'tthink that was needed. I readded it.

**Pull request closed without merging**